source/NPC.cpp

```diff
@@ -672,6 +794,40 @@ NPC NPC::Instantiate(map<string, string> &subs, const System *origin, const Syst
 		result.system = location.PickSystem(origin);
 	if(!result.system)
 		result.system = (isAtDestination && destination) ? destination : origin;
+
+	if(needsWaypoint)
+		result.waypoints.push_back(result.destination);
+	for(const LocationFilter &filter : waypointFilters)
+	{
+		// Each new waypoint calculates "distance X Y" from the previous waypoint / origin.
+		const System *choice = filter.PickSystem(result.waypoints.empty() ?
+				origin : result.waypoints.back());
```

Comment:
If we're doing this, we probably shouldn't be using the same keyword as mission waypoints, which don't care about the order they are defined.

Reply:
Hmmm... I'm not sure what else to call them, though. Any ideas?